src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -349,9 +366,18 @@ void drawing_classifier::init_training(
   std::vector<std::string> gpu_names = training_compute_context_->gpu_names();
   print_training_device(gpu_names);
 
-  // Initialize the neural net. Note that this depends on statistics computed
-  // by the data iterator.
-  nn_spec_ = init_model();
+  // If warm_start enabled, use warmstart CoreML model to
+  // initialize the neural net. Else, call init_model() to
+  // initialize the neural net using xavier.
```

Comment:
Don't we still need to initialize the final layers? We need to append the class-specific layers, like we do for OD

Reply:
Yup, working on that right now

Reply:
that's a good point. I think we only need the weights.

Reply:
I forgot to mention this while I was cooking just now.

Reply:
@nickjong right now I'm deciding between (a) repeating dense1 onwards init code from `init_model()` in the loop on line 372. Or (b) move the coldstart + warmstart model definition inside `init_model()` but this would require passing `warm_start` as an argument and it effects `load_version()` which calls `init_model()`

Reply:
it's better just call init_model first and copy weights from warm start to avoid any incompatible issues of different spec versions.